source/widgets/textbox.c

```diff
@@ -823,6 +826,42 @@ static void textbox_cursor_bkspc(textbox *tb) {
     textbox_cursor_del(tb);
   }
 }
+
+/**
+ * @param tb Handle to the textbox
+ *
+ * Transpose the two characters before the cursor.
+ */
+static void textbox_transpose_chars(textbox *tb) {
+  // Need to have more then 2 characters in front of the cursor.
```

Comment:
The function can also be written with early returns to avoid the nested ifs.

But that's a bit subjective, up to you :)

Reply:
yeah, I was debating this when writing it.  I suspect the compiler turns it into the same code.